akka-docs/src/main/paradox/stream/operators/Source-or-Flow/mapWithResource.md

```diff
@@ -0,0 +1,65 @@
+# mapWithResource
+
+Map elements with the help of a resource that can be opened, transform each element (in a blocking way) and closed.
+
+@ref[Simple operators](../index.md#simple-operators)
+
+## Signature
+
+@apidoc[Flow.mapWithResource](Flow) { scala="#mapWithResource%5BS%2C%20T%5D%28create%3A%20%28%29%20%3D%3E%20S%29%28f%3A%20%28S%2C%20Out%29%20%3D%3E%20T%2C%20close%3A%20S%20%3D%3E%20Option%5BT%5D%29%3A%20Repr%5BT%5D" java="#mapWithResource(akka.japi.function.Creator,akka.japi.function.Function2,akka.japi.function.Function)" }
+1. `create`: Open or Create the resource.
+2. `f`: Transform each element inputs with the help of resource.
+3. `close`: Close the resource, invoked on end of stream or if the stream fails.
+
+## Description
+
+Transform each stream element with the help of a resource.
+The functions are by default called on Akka's dispatcher for blocking IO to avoid interfering with other stream operations. 
+See @ref:[Blocking Needs Careful Management](../../../typed/dispatchers.md#blocking-needs-careful-management) for an explanation on why this is important.
+The resource creation function is invoked once when the stream is materialized and the returned resource is passed to the mapping function for mapping the first element. The mapping function returns a mapped element to emit downstream. The returned T MUST NOT be null as it is illegal as stream element - according to the Reactive Streams specification.
+
+The close function is called only once when the upstream or downstream finished, You can do some clean-up here, and if the returned value is not empty, it will be emitted to the downstream if available, otherwise the value will be dropped.
```

Comment:
Tests show that the cleanup/close also called when it fails and not only for complete/cancel?

Reply:
tests added

Reply:
I meant that this needs to say that the close function is called both on completion _and_ failures

Reply:
I see, will update the document off job